runtime/compiler/control/JITServerCompilationThread.cpp

```diff
@@ -107,16 +107,27 @@ outOfProcessCompilationEnd(
    // If the server is running low on memory, tell clients to reduce the number of active compilation threads
    bool incompleteInfo;
    uint64_t freePhysicalMemorySizeB = compInfoPT->getCompilationInfo()->computeAndCacheFreePhysicalMemory(incompleteInfo);
-   bool serverHasLowMemory = (freePhysicalMemorySizeB != OMRPORT_MEMINFO_NOT_AVAILABLE &&
-       freePhysicalMemorySizeB <= (uint64_t)TR::Options::getSafeReservePhysicalMemoryValue() + 4 * TR::Options::getScratchSpaceLowerBound());
+   JITServer::ServerMemoryState memoryState = JITServer::ServerMemoryState::NORMAL;
+   if (freePhysicalMemorySizeB != OMRPORT_MEMINFO_NOT_AVAILABLE)
+      {
+      size_t numClients = compInfoPT->getCompilationInfo()->getClientSessionHT()->size();
+      uint64_t lowMemoryThreshold = TR::Options::getSafeReservePhysicalMemoryValue() + numClients * TR::Options::getScratchSpaceLowerBound();
```

Comment:
I wonder if dependency on the number of clients can make the lowMemoryThreshold very large when clients do not inform the server before terminating. I would put a hard stop on that (numClients but no more than 16, let's say)